recipes/disopred/addcmdarguments.patch

```diff
@@ -0,0 +1,178 @@
++my ($args, $hits_file, $chk_file) = (undef, "", "");
++if ($mtx == 0) {
++	($hits_file, $chk_file) = map {my $name = join '.', $tmp_base, $_; join '/', $out_dir, $name } ("blast", "chk");
++
++	print "Running PSI-BLAST search ...\n\n";
++	# run psiblast
++	$args = join ' ', $NCBI_DIR."blastpgp", "-i", $fasta_fn, "-d", $SEQ_DB, $PSIBLAST_PAR, "-C", $chk_file, "-o", $hits_file, "\n";
++	system($args) == 0 or die "[$0] ERROR: $args failed: $?\n";
++
++	print "Generating PSSM ...\n\n";
++
++	$args = join ' ', "echo", $chk_file, ">", $tmp_base.".pn", "\n";
++	system($args) == 0 or die "[$0] ERROR: $args failed: $?\n";
++
++	$args = join ' ', "echo", $fasta_fn, ">", $tmp_base.".sn", "\n";
++	system($args) == 0 or die "[$0] ERROR: $args failed: $?\n";
++
++	$args = join ' ', $NCBI_DIR."makemat", "-P", $tmp_base, "\n";
++	system($args) == 0 or die "[$0] ERROR: $args failed: $?\n";
++
++	$mtx_fn = join '/', $out_dir, $tmp_base.".mtx";
++	die "[$0] ERROR: Couldn't find the mtx file $mtx_fn\n" if !-e $mtx_fn;
++} else {
++	print "Re-using pre-computed PSI-BLAST search ...\n\n";
++}
+ 
+ my @exts = ("diso", "diso2", "nndiso", "dnb", "diso", "in_svm_dat", "out_svm_dat", "pbdat");
+ my ($diso_fn, $diso2_fn, $nndiso_fn, $dnb_fn, $diso3_fn, $dat_fn, $svc_fn, $pb_fn) = map { abs_path(join '/', $out_dir, $base.".$_") } @exts;
+@@ -108,12 +128,78 @@
+ 
+ # Remove temporary files
+ print "Cleaning up ...\n\n";
+-$args = join ' ', "rm -f", $hits_file, $chk_file, "error.log", $mtx_fn, <$tmp_base*>, glob("$out_dir/*horiz_d"), $diso2_fn, $nndiso_fn, $dnb_fn, $dat_fn, $svc_fn, "\n";
++my @psiblast_files = ();
++@psiblast_files = ($hits_file, $mtx_fn, $chk_file) if ($mtx == 0);
++$args = join ' ', "rm -f", @psiblast_files, "error.log", <$tmp_base*>, glob("$out_dir/*horiz_d"), $diso2_fn, $nndiso_fn, $dnb_fn, $dat_fn, $svc_fn, "\n";
+ system($args) == 0 or die "[$0] ERROR: $args failed: $?\n";
+ 
+ print join "\n\n", "Finished", "Disordered residue predictions in $diso3_fn", "Protein binding disordered residue predictions in $pb_fn", '';
+ 0;
+ 
++# Process command line arguments
++sub get_arguments {
++
++	my $result = GetOptions ("n=s" => \$NCBI_DIR,
++				"mtx=i" => \$mtx,
++				"d=s" => \$SEQ_DB,
++				"o=s" => \$outdir,
++			        "h"  => sub {&usage;});
++
++	&usage if (!$ARGV[0]);
++
++	unless($mtx){
++		## Check the NCBI directory
++		unless (-d $NCBI_DIR){
++
++			## Look for the NCBI directory
++			my $system = `which blastpgp`;
++			if ($system =~ /(.*)\/blastpgp$/){
++				$NCBI_DIR = $1;
++				$NCBI_DIR =~ s/\s+//g;	
++			}
++			
++			$NCBI_DIR .= '/' if (substr($NCBI_DIR, -1, 1) ne '/');
++
++			unless (-d $NCBI_DIR){
++				print "NCBI directory $NCBI_DIR doesn't exist. Please pass it using\n"; 
++				print "the -n paramater or modifiy the value at the top of the script.\n\n";
++				exit 1;
++			}
++		}
++
++		## Make sure we can find blastpgp & makemat
++		my $psiblast = $NCBI_DIR."/blastpgp";
```

Comment:
That's good to know, especially if you think about windows paths where we would need to replace / by \ !
I coded >= 10 years in Perl but there is still something I don't know :-)